rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -2959,27 +2963,34 @@ error_code sys_fs_get_mount_info(ppu_thread&, vm::ptr<CellFsMountInfo> info, u32
 		sys_fs.todo("sys_fs_get_mount_info special case TODO");
 	}
 
-	const u32 max_len = std::min<u32>(len, 9);
+	const u32 max_len = std::min<u32>(len, std::count_if(mp_array.begin(), mp_array.end(), [](lv2_fs_mount_point* mp){return mp != &g_mp_sys_no_device && mp != &g_mp_sys_dev_usb && mp->is_mounted;}) + std::popcount(g_mp_sys_dev_usb.is_mounted));
 	*out_len = max_len;
 
 	struct mount_info
 	{
-		std::string_view path, filesystem, dev_name;
+		std::string path, filesystem, dev_name;
 		be_t<u32> unk1 = 0, unk2 = 0, unk3 = 0, unk4 = 0, unk5 = 0;
 	};
 
-	static constexpr std::array<mount_info, 9> data
-	{
-		mount_info{.path = "/", .filesystem = "CELL_FS_ADMINFS", .dev_name = "CELL_FS_ADMINFS:", .unk5 = 0x10000000},
-		mount_info{.path = "/app_home", .filesystem = "CELL_FS_DUMMY", .dev_name = "CELL_FS_DUMMY:"},
-		mount_info{.path = "/host_root", .filesystem = "CELL_FS_DUMMY", .dev_name = "CELL_FS_DUMMY:"},
-		mount_info{.path = "/dev_flash", .filesystem = "CELL_FS_FAT", .dev_name = "CELL_FS_IOS:BUILTIN_FLSH1:", .unk5 = 0x10000000},
-		mount_info{.path = "/dev_flash2", .filesystem = "CELL_FS_FAT", .dev_name = "CELL_FS_IOS:BUILTIN_FLSH2:"},
-		mount_info{.path = "/dev_flash3", .filesystem = "CELL_FS_FAT", .dev_name = "CELL_FS_IOS:BUILTIN_FLSH3:"},
-		mount_info{.path = "/dev_hdd0", .filesystem = "CELL_FS_UFS", .dev_name = "CELL_FS_UTILITY:HDD0:"},
-		mount_info{.path = "/dev_hdd1", .filesystem = "CELL_FS_FAT", .dev_name = "CELL_FS_UTILITY:HDD1:"},
-		mount_info{.path = "/dev_bdvd", .filesystem = "CELL_FS_ISO9660", .dev_name = "CELL_FS_IOS:PATA0_BDVD_DRIVE"},
-	};
+	std::vector<mount_info> data;
+
+	for (auto mp : mp_array)
+	{
+		if (mp == &g_mp_sys_dev_usb)
+		{
+			for (int i = 0; i < 8; i++)
+			{
+				if ((mp->is_mounted >> i) & 1U)
+				{
+					data.push_back(mount_info{.path = fmt::format("%s%03d", mp->root, i), .filesystem = mp->file_system.data(), .dev_name = fmt::format("%s%03d", mp->device, i)});
+				}
+			}
+		}
+		else if (mp != &g_mp_sys_no_device && mp->is_mounted)
```

Comment:
you can optimize this whole loop by just continuing at the beginning if mp->is_mounted == 0

Reply:
Done.